NBitcoin/Utils.cs

```diff
@@ -699,7 +698,21 @@ public static IPEndPoint ParseIpEndpoint(string endpoint, int defaultPort)
 			}
 			catch(FormatException)
 			{
+#if !WINDOWS_UWP
 				address = Dns.GetHostEntry(ip).AddressList[0];
+#else
+				foreach(HostName localHostName in NetworkInformation.GetHostNames())
+				{

```

Comment:
I don't understand that, it seems to change completely the logic of ParseIpEndpoint by returning localhost everytimes an IP is not well formed.